File: java_solutions/1290.convert-binary-number-in-a-linked-list-to-integer.java
/*
 * @lc app=leetcode id=1290 lang=java
 *
 * [1290] Convert Binary Number in a Linked List to Integer
 */

class Solution {
    public int getDecimalValue(ListNode head) {
        return Stream.iterate(head, node -> node != null, node -> node.next)
            .map(node -> node.val)
            .reduce(0, (accumulator, bit) -> (accumulator << 1) | bit);
    }
}

File: kotlin_solutions_1/1290.convert-binary-number-in-a-linked-list-to-integer.kt
/*
 * @lc app=leetcode id=1290 lang=kotlin
 *
 * [1290] Convert Binary Number in a Linked List to Integer
 */

class Solution {
    fun getDecimalValue(head: ListNode?): Int = generateSequence(head) { it.next }.fold(0) { acc, node -> (acc shl 1) or node.`val` }
}